### PR TITLE
NIFI-13944: Upgrade Redis to 5.2.0

### PR DESCRIPTION
**Pull request by mattyb149**
# Summary

[NIFI-13944](https://issues.apache.org/jira/browse/NIFI-13944) This PR upgrades the Redis library to 5.2.0 along with the corresponding Spring Redis version, and excludes the banned org.json:json dependency in favor of Ted Dunning's version.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by joewitt** [comment]
Seems to build.   But the bundled dependencies of Spring seem considerable and likely avoidable.  The LICENSE/NOTICE is unchanged but clearly does need to change.

Seems like at least a few modules also included that are likely in a parent class loader.  Reviewing further but this is a good example of a dependency where maintenance jumps (major lines no less) are much more than bumping versions.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the upgrade @mattyb149, this looks good from my perspective.

@joewitt, reviewing the current Redis NAR bundles, they contain roughly the same set of Spring Framework dependencies, most of which are transitive through Spring Data Redis. From that perspective, I think the upgrade works as it stands, although it does highlight the potential need for further review in a separate issue. It is worth noting, however, that only the Framework NAR contains the Spring libraries, so these would not be inherited from the `nifi-standard-shared-nar`. I will defer to you for further review.

**Comment by exceptionfactory** [comment]
Although for clarification, it looks like the `NOTICE` files are outdated, so I agree they should be updated to reflect current contents and versions.

File: nifi-extension-bundles/nifi-redis-bundle/nifi-redis-extensions/pom.xml [hunk 1]
```diff
@@ -62,6 +62,17 @@
             <artifactId>jedis</artifactId>
             <version>${jedis.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.json</groupId>
+                    <artifactId>json</artifactId>
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-redis-bundle/nifi-redis-extensions/pom.xml:68]: This version of the `json` librar [20240303](https://central.sonatype.com/artifact/org.json/json/20240303) is actually licensed under the Public Domain, not the historically banned license. For that reason, the exclusion is not necessary and the banned dependency settings can be adjusted.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
As discussed in an offline conversation, the license and notice files contain more information than needed as dependencies with the `provided` scope should not be referenced.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
This looks closer to completion, but the information in `NOTICE` looks like it reflects what is in `nifi-redis-service-api-nar`, not what is in `nifi-redis-nar`, so both NOTICE files need to be updated.

File: pom.xml [hunk 2]
```diff
@@ -853,7 +853,7 @@
                                         <!-- JUnit 5 is the preferred test framework -->
                                         <exclude>org.testng:testng</exclude>
                                         <!-- Cat-X Deps -->
-                                        <exclude>org.json:json:*:*:compile</exclude>
+                                        <exclude>org.json:json:[,20240303]:compile</exclude>
```
Comment (exceptionfactory) [inline pom.xml:856]: Reviewing the [Maven Version Ranges](https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html), it looks like this should use parentheses to indicate `x < 20240303`:
```suggestion
                                        <exclude>org.json:json:(,20240303):compile</exclude>
```

**Review by exceptionfactory — APPROVED** [review]
Thanks for the updates @mattyb149. On further review, I noticed several additional updates needed, which I pushed in the latest commit. +1 merging

**Pull request merged**